lib/google/ads/google_ads/google_ads_client.rb

```diff
@@ -140,36 +131,73 @@ def service(name, version = default_api_version)
             end
             headers[:"login-customer-id"] = login_customer_id.to_s  # header values must be strings
           end
-          return class_to_return.new(
-            credentials: get_updater_proc(),
-            metadata: headers,
-            exception_transformer: ERROR_TRANSFORMER
-          )
+
+          if logger
+            logging_interceptor = Google::Ads::GoogleAds::LoggingInterceptor.new(logger)
+          end
+
+          if name.nil?
+            Factories::Services.new(
```

Comment:
Should we instantiate one of these each time, or can we cache it and just reference the one we have?

Also, you need to build some infrastructure to support multiple versions of the factories at once, and use the correct version of them.

Reply:
Creating one of these each time is probably safer for the time being, once this is more established and also once we investigate GRPC connection pooling further, we'll have a clearer idea of what to do here.